source/docs/flow_control.md

```diff
@@ -278,4 +278,59 @@ called on creation.
 
 ### HTTP implementation details
 
-TODO(alyssawilk) implement and document.
+HTTP flow control is extremely similar to HTTP/2 flow control, with the main exception being that
+the method used to halt the flow of downstream/upstream data is to disable reads on the underlying
+Network::Connection.  As the TCP data stops being consumed the peer will eventually fill their
+congestion window and stop sending.
+
+As with HTTP/2, a given stream may end in a state where the connection has had
+`readDisable(true)` called.   When a new stream is created on that connection
+for a subsequent request, any outstanding `readDisable(true)` calls are unwound
+in `Http::Http1::ConnectionImpl::newStream()`.
+
+Filter and network backups are identical in the HTTP and HTTP/2 cases and are
+documented above.  Codec backup is slightly different and is documented below.
+
+# HTTP codec downstream send buffer
+
+As with the HTTP/2 codec recv buffer, the HTTP codec send buffer is only expected to have data pass
+through it.  It should never back up.  Still, in the event it does, the high
```

Comment:
Ah, so I planned on doing that way bck on whatever recv/send buffer didn't really need checks. Unfortunately because we don't have hard limits you can't have an assert - some large swathe of data can come on through and fill your buffer for just one cycle and then it drains "immediately" (but not really immediately)